test/e2e/user_configured_cabundle_test.go

```diff
@@ -0,0 +1,692 @@
+//     is present in the `ingress-ca-bundle`.
+//  3. On updating `admin-ca-bundle` or `service-ca-bundle`, operator detects changes
+//     and updates `ingress-ca-bundle` and router reloads HAProxy.
+func TestIngressCABundle(t *testing.T) {
+	var (
+		ctx = context.TODO()
+
+		testServerNamespace = "ingress-ca-bundle-test"
+		openshiftServerCA   = types.NamespacedName{
+			Name:      "openshift-service-ca.crt",
+			Namespace: "openshift-config",
+		}
+		serverCABundleConfigMapName = types.NamespacedName{
+			Name:      "test-server-ca-bundle",
+			Namespace: testServerNamespace,
+		}
+
+		serviceName1      = "hello-world-1"
+		svcTLSSecretName1 = types.NamespacedName{
+			Name:      serviceName1,
+			Namespace: testServerNamespace,
+		}
+		svcDeploymentName1 = types.NamespacedName{
+			Name:      serviceName1,
+			Namespace: testServerNamespace,
+		}
+		svcLabels1 = map[string]string{
+			"app": serviceName1,
+		}
+		svcPort1 = int32(8082)
+		svcName1 = types.NamespacedName{
+			Name:      serviceName1,
+			Namespace: testServerNamespace,
+		}
+		routeName1 = types.NamespacedName{
+			Name:      serviceName1,
+			Namespace: testServerNamespace,
+		}
+
+		serviceName2      = "hello-world-2"
+		svcTLSSecretName2 = types.NamespacedName{
+			Name:      serviceName2,
+			Namespace: testServerNamespace,
+		}
+		svcDeploymentName2 = types.NamespacedName{
+			Name:      serviceName2,
+			Namespace: testServerNamespace,
+		}
+		svcLabels2 = map[string]string{
+			"app": serviceName2,
+		}
+		svcPort2 = int32(8083)
+		svcName2 = types.NamespacedName{
+			Name:      serviceName2,
+			Namespace: testServerNamespace,
+		}
+		routeName2 = types.NamespacedName{
+			Name:      serviceName2,
+			Namespace: testServerNamespace,
+		}
+	)
+
+	svcCACert1, svcCAKey1, err := generateCAKeyPair("operator-e2e-1")
+	if err != nil {
+		t.Fatalf("failed to generate first server CA certificate: %v", err)
+	}
+	svcCert1, svcKey1, err := generateServerCertificate(svcCACert1, svcCAKey1, serviceName1, []string{serviceName1, fmt.Sprintf("%s.%s.svc", serviceName1, testServerNamespace)})
+	if err != nil {
+		t.Fatalf("failed to generate first server certificate: %v", err)
+	}
+	clientCert, clientKey, err := generateClientCertificate(svcCACert1, svcCAKey1, "test-client")
+	if err != nil {
+		t.Fatalf("failed to generate client certificate: %v", err)
+	}
+	svcCACert2, svcCAKey2, err := generateCAKeyPair("operator-e2e-2")
+	if err != nil {
+		t.Fatalf("failed to generate first server CA certificate: %v", err)
+	}
+	svcCert2, svcKey2, err := generateServerCertificate(svcCACert2, svcCAKey2, serviceName2, []string{serviceName2, fmt.Sprintf("%s.%s.svc", serviceName2, testServerNamespace)})
+	if err != nil {
+		t.Fatalf("failed to generate second server certificate: %v", err)
+	}
+
+	// Create the configmap with the custom CA certificates which will be
+	// used for verifying services' certificates.
+	adminCABundleConfigMapName := controller.AdminCAConfigMapName()
```

Comment:
We've been moving towards using `t.Log` more for these sorts of comments in E2E tests to make it easier to trace the test steps when something goes wrong:
```suggestion
	adminCABundleConfigMapName := controller.AdminCAConfigMapName()
	t.Logf("creating %s configmap", adminCABundleConfigMapName)

Reply:
suggestion incorporated.